javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +519,375 @@ object Character {
   @inline def compare(x: scala.Char, y: scala.Char): Int =
     x - y
 
+  // UnicodeBlock
+
+  class Subset protected (name: String) {
+    override def equals(that: Any): scala.Boolean = super.equals(that)
+    override def hashCode: scala.Int              = super.hashCode
+    override def toString(): String               = name
+  }
+
+  final class UnicodeBlock protected (name: String) extends Subset(name) {
+    private var start: Int = _
+    private var end: Int = _
+    private def this(name: String, start: Int, end: Int) = {
+      this(name)
+      this.start = start
+      this.end = end
+    }
+  }
+
+  object UnicodeBlock {
+    private[this] val totalCodeBlocks: Int = 220
+    private[this] var currentBlockIndex: Int = 0
+    private[this] val allBlocks: Array[UnicodeBlock] = new Array[UnicodeBlock](totalCodeBlocks)
```

Comment:
Consider using an `ArrayList` so you don't need to hardcode the count? (I doubt it matter for the overall speed)